plugins/module_utils/network/sonic/config/dhcp_snooping/dhcp_snooping.py

```diff
@@ -0,0 +1,641 @@
+
+        commands = get_diff(want, have, test_keys=self.test_keys)
+        requests = self.get_modify_requests(commands)
+
+        if commands and len(requests) > 0:
+            commands = update_states(commands, "merged")
+        else:
+            commands = []
+
+        return commands, requests
+
+    def _state_deleted(self, want, have, afis):
+        """ The command generator when state is deleted
+
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        requests = []
+        if not have or not have.get('afis'):
+            # nothing that could be deleted
+            commands = []
+        elif not want or not want.get('afis'):
+            # want is empty, meaning want to delete all config
+            # afis parameter only stores the on device config at this point
+            commands, requests = self.get_delete_all_have_requests(afis)
+        else:
+            # some mix of settings specified in both
+            commands, requests = self.get_delete_specific_requests(afis)
+
+        if commands and len(requests) > 0:
+            commands = update_states(commands, "deleted")
+        else:
+            commands = []
+        return commands, requests
+
+    def _state_overridden(self, want, have, afis):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        if not want:
+            return commands, requests
+
+        # Determine if there is any configuration specified in the playbook
+        # that is not contained in the current configuration.
+        diff_requested = get_diff(want, have, self.test_keys)
+
+        # Determine if there is anything already configured that is not
+        # specified in the playbook.
+        diff_unwanted = get_diff(have, want, self.test_keys)
+
+        # Idempotency check: If the configuration already matches the
+        # requested configuration with no extra attributes, no
+        # commands should be executed on the device.
+        if not diff_requested and not diff_unwanted:
+            return commands, requests
+
+        used_commands_per_afi = []
+        commands = []
+
+        for diff_unwanted_afi in diff_unwanted.get("afis", []):
+            # enabled and verify_mac can't be deleted from config, only set to default. 
+            #   so in the case they appear in both the "need to delete" and "need to change", keeping in both results in double requests
+            if "enabled" in diff_unwanted_afi and "enabled" in diff_requested:
+                del diff_unwanted_afi["enabled"]
+            if "verify_mac" in diff_unwanted_afi and "verify_mac" in diff_requested:
+                del diff_unwanted_afi["verify_mac"]
```

Comment:
'enabled' and 'verify_mac' are sub options of 'afis' list and their presence needs to be checked in the list elements within 'diff_requested' similar to 'diff_unwanted_afi' in the current implementation.